pkg/materialize/connection_aws.go

```diff
@@ -0,0 +1,183 @@
+	b.assumeRoleArn = s
+	return b
+}
+
+func (b *ConnectionAwsBuilder) AssumeRoleSessionName(s string) *ConnectionAwsBuilder {
+	b.assumeRoleSessionName = s
+	return b
+}
+
+func (b *ConnectionAwsBuilder) Validate(validate bool) *ConnectionAwsBuilder {
+	b.validate = validate
+	return b
+}
+
+func (b *ConnectionAwsBuilder) Create() error {
+	q := strings.Builder{}
+	q.WriteString(fmt.Sprintf(`CREATE CONNECTION %s TO AWS`, b.QualifiedName()))
+
+	w := []string{}
+	if b.endpoint != "" {
+		o := fmt.Sprintf(` ENDPOINT = %s`, QuoteString(b.endpoint))
+		w = append(w, o)
+	}
+	if b.region != "" {
+		o := fmt.Sprintf(` REGION = %s`, QuoteString(b.region))
+		w = append(w, o)
+	}
+
+	if b.accessKeyId.Text != "" {
+		o := fmt.Sprintf(` ACCESS KEY ID = %s`, QuoteString(b.accessKeyId.Text))
+		w = append(w, o)
+	} else if b.accessKeyId.Secret.Name != "" {
+		o := fmt.Sprintf(` ACCESS KEY ID = SECRET %s`, b.accessKeyId.Secret.QualifiedName())
+		w = append(w, o)
+	}
+	if b.secretAccessKey.Name != "" {
+		o := fmt.Sprintf(` SECRET ACCESS KEY = SECRET %s`, b.secretAccessKey.QualifiedName())
+		w = append(w, o)
+	}
+	if b.sessionToken.Text != "" {
+		o := fmt.Sprintf(` SESSION TOKEN = %s`, QuoteString(b.sessionToken.Text))
+		w = append(w, o)
+	} else if b.sessionToken.Secret.Name != "" {
+		o := fmt.Sprintf(` SESSION TOKEN = SECRET %s`, b.sessionToken.Secret.QualifiedName())
+		w = append(w, o)
+	}
+
+	if b.assumeRoleArn != "" {
+		o := fmt.Sprintf(` ASSUME ROLE ARN = %s`, QuoteString(b.assumeRoleArn))
+		w = append(w, o)
+	}
+	if b.assumeRoleSessionName != "" {
+		o := fmt.Sprintf(` ASSUME ROLE SESSION NAME = %s`, QuoteString(b.assumeRoleSessionName))
+		w = append(w, o)
+	}
+
+	if !b.validate {
+		w = append(w, " VALIDATE = false")
+	}
+
+	f := strings.Join(w, ", ")
+	q.WriteString(fmt.Sprintf(` WITH (%s)`, f))
+	return b.ddl.exec(q.String())
+}
+
+type ConnectionAwsParams struct {
+	ConnectionId            sql.NullString `db:"id"`
+	ConnectionName          sql.NullString `db:"connection_name"`
+	SchemaName              sql.NullString `db:"schema_name"`
+	DatabaseName            sql.NullString `db:"database_name"`
+	Endpoint                sql.NullString `db:"endpoint"`
+	Region                  sql.NullString `db:"region"`
+	AccessKeyId             sql.NullString `db:"access_key_id"`
+	AccessKeyIdSecretId     sql.NullString `db:"access_key_id_secret_id"`
+	SecretAccessKeySecretId sql.NullString `db:"secret_access_key_secret_id"`
+	SessionToken            sql.NullString `db:"session_token"`
+	SessionTokenSecretId    sql.NullString `db:"session_token_secret_id"`
+	AssumeRoleArn           sql.NullString `db:"assume_role_arn"`
+	AssumeRoleSessionName   sql.NullString `db:"assume_role_session_name"`
+	Comment                 sql.NullString `db:"comment"`
+	Principal               sql.NullString `db:"principal"`
+	OwnerName               sql.NullString `db:"owner_name"`
+}
+
+var connectionAwsQuery = NewBaseQuery(`
+	SELECT
```

Comment:
Haven't been able to fully vet this query since the system table is not in the image yet. This is based on the [privatelink query](https://github.com/MaterializeInc/terraform-provider-materialize/blob/main/pkg/materialize/connection_aws_privatelink.go#L62)